### PR TITLE
Switch imports from test_api to focused libaries

### PR DESCRIPTION
**Pull request by natebosch**
Move imports of API available in `hooks.dart` or `scaffolding.dart` to use those more narrow libraries.

Move imports of APIs from `package:matcher` to import directly. The next major version of `test_api` will remove the exports of `matcher` APIs.

### CODE REVIEW

File: packages/flutter_test/lib/src/_binding_io.dart [hunk 1]
```diff
@@ -10,7 +10,7 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter/services.dart';
 import 'package:path/path.dart' as path;
 // ignore: deprecated_member_use
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/_binding_io.dart:12]: Looks like scaffolding.dart isn't deprecated, can we remove this ignore then? (here and elsewhere)

https://github.com/dart-lang/test/blob/master/pkgs/test_api/lib/scaffolding.dart

Comment (natebosch) [reply packages/flutter_test/lib/src/_binding_io.dart:12]: It's deprecated in an older version of `test_api`, I was still getting diagnostics but maybe I just hadn't updated my local checkout to the lasted pub package versions.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by natebosch** [comment]
I don't understand the failure on "Windows customer_testing" - it looks like it may be some infrastructure failure?

**Comment by christopherfujino** [comment]
> I don't understand the failure on "Windows customer_testing" - it looks like it may be some infrastructure failure?

customer_testing compares the registry against the branch point from master. If this was fixed upstream you'd need to rebase (not merge) upstream master to pick up the fix.

**Comment by natebosch** [comment]
> you'd need to rebase (not merge)

If it fails again I will try a rebase. Is there any discussion of fixing this so it isn't a requirement to rewrite shared history?

**Comment by christopherfujino** [comment]
> > you'd need to rebase (not merge)
> 
> If it fails again I will try a rebase. Is there any discussion of fixing this so it isn't a requirement to rewrite shared history?

not sure.

**Comment by natebosch** [comment]
It passed this time, so it may have been a flake.

**Comment by natebosch** [comment]
I'm not sure whether "Google testing" is still doing any work, it says "Presubmit Test Passed"...

**Comment by goderbauer** [comment]
There's a bug that sometimes Frob doesn't sync back a successful test run. I manually set it to green.

**Pull request merged**